mantle/platform/api/azure/groups.go

```diff
@@ -41,18 +42,34 @@ func (a *API) CreateResourceGroup(prefix string) (string, error) {
 }
 
 func (a *API) TerminateResourceGroup(name string) error {
-	resp, err := a.rgClient.CheckExistence(name)
+	resp, err := a.rgClient.CheckExistence(context.Background(), name, nil)
 	if err != nil {
 		return err
 	}
-	if resp.StatusCode != 204 {
+	if !resp.Success {
 		return nil
 	}
 
-	_, err = a.rgClient.Delete(name, nil)
+	_, err = a.rgClient.BeginDelete(context.Background(), name, nil)
```

Comment:
Likewise.

Reply:
hmm - this is one where I would think we don't care as much.. i.e. if the resource group takes 5 minutes to delete we don't care (as long as it does get cleaned up) and let our process continue on.

Reply:
I guess the only reason we would care maybe is if we try to then create a resource group again right after with the same name, though that's unlikely?

Reply:
Yeah, that's fair.  I seem to recall that resource group deletion can take several minutes, and you're right that the caller is unlikely to have a dependency on the RG being gone.  Let's skip this one, but maybe document that the function doesn't wait for deletion?

Reply:
done